src/agent/main.cpp

```diff
@@ -47,7 +43,7 @@
 #include <openthread/logging.h>
 #include <openthread/platform/radio.h>
 
-#if __ANDROID__ && OTBR_CONFIG_ANDROID_PROPERTY_ENABLE
+#if OTBR_PLATFORM_ANDROID
```

Comment:
All otbr related configurations are with the prefix `OTBR_ENABLE`. Should we define it as `OTBR_ENABLE_PLATFORM_ANDROID`